### PR TITLE
Pc issue 8: Add Edit Page where Dining Commons can be edited

### LINKED ISSUE #8: Create an edit page
# User Story

As a user, I can go to a page where I can edit a dining commons database entry so that I can make needed changes to the records.

# AC

- [ ] There is a page where I can change the values of all fields except for the `code` field
- [ ] When I click `Update`, those fields actually get changed in the database
- [ ] When I click `Update` I am rerouted to the index page 
- [ ] When I click `Cancel` no changes are made and I'm rerouted to whatever page I just came from

### PR DESCRIPTION
**Pull request by pconrad**
# Overview

In this PR, we add the edit page for DiningCommons.

For now, you have to go to this page manually; there is no button for it yet.  We'll do that in the next PR.

So to test, you put in a url such as: <http://localhost:8080/diningCommons/edit/ortega>

You'll see a page like this one, where you can edit the `ortega` dining commons:

image

# Issues Addressed

closes #8 

# Test Plan (for interactive testing)

1. Go to the index page and make sure you have a DiningCommons record
2. Go to this url, where you have the `code` of a dining commons record in place of `ortega`
3. Try editing the record
4. When you click `Cancel`, it should not update
5. When you click `Update`, it should update and return you to the index page (i.e. the list of commons)

### CODE REVIEW

**Review by andrewpengucsb — APPROVED** [review]
LGTM!

File: frontend/src/main/pages/DiningCommons/DiningCommonsCreatePage.js [hunk 1]
```diff
@@ -1,13 +1,53 @@
 import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+import DiningCommonsForm from "main/components/DiningCommons/DiningCommonsForm";
+import { Navigate } from 'react-router-dom'
+import { useBackendMutation } from "main/utils/useBackend";
+import { toast } from "react-toastify";
+
+export default function DiningCommonssCreatePage() {
```
Comment (andrewhlu) [inline frontend/src/main/pages/DiningCommons/DiningCommonsCreatePage.js]: same misspelling here, but I think this will get resolved if you merge #14 first

**Review by andrewhlu — APPROVED** [review]
LGTM!

File: frontend/src/main/pages/DiningCommons/DiningCommonsCreatePage.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/main/pages/DiningCommons/DiningCommonsCreatePage.js]: Yes, rebasing fixed that.

**Pull request merged**